### PR TITLE
Move file selector

### PR DESCRIPTION
**Pull request by christinem**
This PR moved the file selector into the annotations menu. It currently removes the "download files" functionality, as I removed the original file selector bar. It will be added back in the next pull request.

### CODE REVIEW

File: app/views/results/common/_file_selector_dropdown.html.erb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        first_file_path = null;
+        <% else %>
+        first_file_id = <%= files[0].id %>;
+        first_file_path = '<%= files[0].path %>/<%= files[0].filename %>';
+        <% end %>
+      </script>
+
+
+      <!-- arrange the files list into a nested hashtable for displaying in a recursive menu -->
+      <% outermost_dir = Hash.new %>
```
Comment (david-yz-liu) [inline app/views/results/common/_file_selector_dropdown.html.erb]: @christinem make this a helper function (in ResultsHelper)

**Comment by christinem** [comment]
@david-yz-liu Done. Also added a parallel menu like the annotation menu for the student and ta "released" views, with just the file selector, to fix the bug from yesterday.

File: app/assets/stylesheets/grader.scss [hunk 2]
```diff
@@ -235,14 +209,51 @@
...
+.download_file {
+  background: $light-grey;
+  border-radius: $radius;
+  display: table-cell;
+  vertical-align: top;
+  float: left;
+  line-height: 16.25px;
+}
+
+#file_selector_seperator {
```
Comment (david-yz-liu) [inline app/assets/stylesheets/grader.scss]: There's a typo in "seperator" here

File: app/helpers/results_helper.rb [hunk 3]
```diff
@@ -37,4 +37,23 @@ def can_show_test_results_tab?(assignment)
   def can_show_feedback_files_tab?(submission)
     not submission.feedback_files.empty?
   end
+
+  def create_nested_files_hash_table(files)
+    # arrange the files list into a nested hashtable for displaying in a recursive menu
+    outermost_dir = Hash.new
+    files.each do |file|
+      innermost_dir = outermost_dir
+      innermost_dir["files"] = Array.new
```
Comment (david-yz-liu) [inline app/helpers/results_helper.rb]: Try to use symbols instead of strings for keys.

File: app/views/results/marker/_marker_panes.html.erb [hunk 4]
```diff
@@ -24,6 +24,18 @@
           <div id='sel_box'></div>
 
           <div id='annotation_menu'>
+
+            <%= render partial: 'results/common/file_selector_dropdown',
+                       locals: { files: files,
+                                 can_download: true } %>
+            <% if can_show_feedback_files_tab?(submission) %>
+                <div id='feedback_files_controls'>
```
Comment (david-yz-liu) [inline app/views/results/marker/_marker_panes.html.erb]: While you're at it, can you please move this part to the feedback files tabs?

**Comment by reidka** [comment]
I tested this out.  Looks great!

**Comment by david-yz-liu** [comment]
@christine I will merge this in. However, this is one UI flaw: the dropdown doesn't "reset" after a file has been selected and the page reloads.

**Comment by christinem** [comment]
@david-yz-liu thanks David. I can come back to it and make that UI fix in another PR.

**Pull request merged**